geoips_clavrx/plugins/modules/readers/clavrx_hdf4.py

```diff
@@ -20,19 +21,44 @@
 
 with import_optional_dependencies(loglevel="info"):
     """Attempt to import a package and print to LOG.info if the import fails."""
-    from pyhdf.SD import SD, SDC
     from pyhdf.error import HDF4Error
+    from pyhdf.SD import SD, SDC
 
 interface = "readers"
 family = "standard"
 name = "clavrx_hdf4"
 
 
-def parse_metadata(metadatadict):
-    """Parse metadata."""
-    metadata = {}
-    for ii in metadatadict.keys():
-        metadata[ii] = metadatadict[ii]
+SCALED_ATTRIB = "SCALED"
+SCALE_FACTOR_ATTRIB = "scale_factor"
+ADD_OFFSET_ATTRIB = "add_offset"
+SCALED_FLAG = 1
+FILL_ATTRIB = "_FillValue"
+MISSING_ATTRIB = "actual_missing"
+DIMS = ["y", "x"]
+
+
+SCALING_ATTRIBS = [
+    SCALED_ATTRIB,
+    SCALE_FACTOR_ATTRIB,
+    ADD_OFFSET_ATTRIB,
+]
+
+
+def parse_metadata(metadata_in):
+    """Parse and correct metadata.
+
+    Parameters
+    ----------
+    metadata_in : dict
+        Input metadata dictionary.
+
+    Returns
+    -------
+    dict
+        Parsed and modified metadata.
+    """
+    metadata = dict(metadata_in)
 
     # Map GOES-RU-IMAGER to "abi" GeoIPS sensor name
```

Comment:
Why is this only required for ABI?

Reply:
I don't know; that's from Songs previous work. I figured it was some kind of historical/anomalous thing